src/qca/utils/utils.py

```diff
@@ -257,7 +267,8 @@ def circuit_estimate(
                 subcircuit_counts[gate_type][0] += 1
             else:
                 t0 = time.perf_counter()
-                decomposed_circuit = cirq.Circuit(cirq.decompose(operation))
+                # decomposed_circuit = cirq.Circuit(cirq.decompose(operation))
+                decomposed_circuit = complete_decomposition(cirq.Circuit(operation))
```

Comment:
same thing here about my previous comment, again not sure if required but shouldn't N be an argument for the function for decomposition?

Reply:
Refer to previous comment